### PR TITLE
feat: allow creating workspaces with an empty repository list

### PR DESCRIPTION
**Pull request by Teo-ShaoWei**
## Context

Users may want to create workspaces without any repositories, using the shell's working directory as the workspace `pwd`. Previously, empty repo lists were rejected at multiple validation layers.

When `hasResolvedInitialRepoDefaults` resolved after the user had already clicked "Continue" with no repos, the initialisation `useEffect` would fire (since `hasInitializedStep` was still `false`) and reset `isSelectingRepos` back to `true`, yanking the user from the chat step back to the repo picker. This was previously invisible because `showChatStep` required `hasSelectedRepos`.

## Changes

- Backend — remove empty-repos validation from:
  - `create_and_start_workspace` API route
  - `start_workspace` MCP tool
  - `start_execution` in the services container
  - `workspace_repo_inputs` in the local container
- `WorkspaceManager`:
  - `create_workspace` now creates the workspace directory even with no repos
  - `ensure_workspace_exists` returns early (after ensuring the directory exists) when repos are empty
- Frontend:
  - `CreateModeRepoPickerBar` — allow "Continue" with no repos selected
  - `CreateChatBoxContainer` — allow submission with no repos, remove the "at least one repository" error, show `No repositories` in the repo summary label
  - `CreateChatBoxContainer` — set `hasInitializedStep` to `true` when the user explicitly clicks "Continue", preventing the late-resolving effect from resetting the step

Contributes to #3061

---

> [!NOTE]
> **Medium Risk**
> Enables workspace creation/execution flows to proceed with an empty repo list, which changes assumptions in container/worktree and UI flows and could expose edge cases around working directories, branch selection, and cleanup.
> 
> **Overview**
> Allows creating and starting workspaces with **zero repositories** by removing empty-repo validation from the API route (`create_and_start_workspace`), MCP tool (`start_workspace`), and container/services layers.
> 
> Updates `WorkspaceManager` to handle empty repo sets by still creating/ensuring the workspace directory (and returning early from `ensure_workspace_exists` when no repos). The web create flow is adjusted to let users continue/submit without selecting repos, removes the “at least one repository” error, and displays a `No repositories` summary label.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9431c1dcd975e237ca1497d2c2e927e9ac56f8e0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/web-core/src/shared/components/CreateChatBoxContainer.tsx [hunk 1]
```diff
@@ -77,8 +77,8 @@ export function CreateChatBoxContainer({
     hasResolvedInitialRepoDefaults,
   ]);
 
-  const showRepoPickerStep = !hasSelectedRepos || isSelectingRepos;
-  const showChatStep = hasSelectedRepos && !isSelectingRepos;
+  const showRepoPickerStep = isSelectingRepos;
```
[the remark this answers is not in the document]
Comment (Teo-ShaoWei) [reply packages/web-core/src/shared/components/CreateChatBoxContainer.tsx:80]: Good catch!

[the remark this answers is not in the document]
Comment (Teo-ShaoWei) [reply packages/web-core/src/shared/components/CreateChatBoxContainer.tsx:80]: Ok updated.

**Comment by Juanlucasbg** [comment]
## Aggressive review summary — PR #3316

64-line feature across 7 files removing the "at least one repository required" check from 5 sites and adjusting the create-workspace UI to allow empty repo lists. Verdict: **clean — recommend merge with one cleanup NIT**.

### Coherence check

The PR removes the empty-repos guard at every layer where it surfaced:

1. `crates/local-deployment/src/container.rs:163` — "Workspace has no repositories configured" → removed; downstream `find_repos_for_workspace` returns empty `Vec`.
2. `crates/mcp/src/task_server/tools/task_attempts.rs:103` — MCP `start_workspace` 400 → removed.
3. `crates/server/src/routes/workspaces/create.rs:228` — REST 400 → removed.
4. `crates/services/src/services/container.rs:1141` — pre-execution HEAD capture early-return → removed; the loop that follows is a no-op over an empty vec, safe.
5. `crates/workspace-manager/src/workspace_manager.rs:294-298` — `create_workspace_dir_with_repos` early-return → removed.
6. `workspace_manager.rs:373-380` — second function gracefully creates the empty dir (`tokio::fs::create_dir_all`) and returns `Ok(())` instead of erroring.

UI changes drop `hasSelectedRepos` from the canSubmit computation, the dropzone disable, and the chat-box `disabled` prop, and replace the empty-state copy with `'No repositories'`. The repo-picker step is now optional in the flow.

### Findings

- **Structural — NIT**: After this PR, no code path emits `WorkspaceError::NoRepositories`. The variant at `crates/workspace-manager/src/workspace_manager.rs:54` and the two error-mapper arms at `crates/server/src/error.rs:132` and `crates/local-deployment/src/container.rs:141` become dead code. Consider removing in this PR or filing a follow-up cleanup.
- **Adversarial**: 0 HIGH. Empty-workspace happy path: agent boots with an empty worktree dir; downstream loops over `Vec<Repo>` are vacuously safe. No null-pointer / divide-by-zero / panic surfaces introduced.
- **Security**: 0 HIGH. No new auth surface; the empty workspace is local-only and doesn't accept network input differently than the populated case.
- **UX consistency**: 0 HIGH. The submit button is now enabled even with zero repos, which is the explicit intent.
- **Conventions**: PASS.

### NITs (non-blocking)

- **Tests**: `CLAUDE.md` says "Add tests for new logic and edge cases." A unit test asserting `WorkspaceManager::create_workspace_dir_with_repos(...)` succeeds with `repos: &[]` — and a corresponding API-level test for `POST /workspaces` with `repositories: []` — would lock the new contract in place.
- **Docs**: `crates/server/src/routes/workspaces/create.rs` likely has OpenAPI / type comments that say "at least one" — update them.
- **What does the user see post-create?** The empty-repos workspace shows up in the kanban with no diff, no branch, no commits. Worth a one-line note in the UI explaining why the empty card has no metadata, or a more deliberate "scratchpad workspace" badge so users understand the state isn't a bug.

### Verdict

Approve.

— Reviewed by automated single-pass review (consistent-stack-removal feature; full 4-tool battery skipped — change is symmetric across layers and the diff is mechanical).
